readthedocs/api/v3/views.py

```diff
@@ -139,8 +137,22 @@ class ProjectsViewSetBase(
         "teams",
     ]
 
+    def get_permissions(self):
+        # Create and list are actions that act on the current user.
+        if self.action in ("create", "list"):
+            permission_classes = [IsAuthenticated]
+        # Actions that change the state of the project require admin permissions on the project.
+        elif self.action in ("update", "partial_update", "destroy"):
+            permission_classes = [IsAuthenticated & IsProjectAdmin]
+        # Any other action is read-only.
+        else:
+            permission_classes = [ReadOnlyPermission]
+        return [permission() for permission in permission_classes]
+
     def get_view_name(self):
         # Avoid "Base" in BrowseableAPI view's title
+        if self.name:
+            return self.name
```

Comment:
If there is a name, that should always take precedence.

https://github.com/encode/django-rest-framework/blob/8e304e1adbb0f99f91a15ed3abd379104bba3b89/rest_framework/views.py#L29-L32

Otherwise, we end with something like `Projects None`

https://app.readthedocs.org/api/v3/projects/stsewd-demo/superproject/